### ISSUE TITLE
class-or-instance part of Java interop calls isn't instrumented

### ISSUE THREAD

**Issue by camsaul** [cloverage/cloverage#306]
I just noticed this while poking around something unrelated.

Java interop forms when fully expanded are either of the syntax

```clj
(. class-or-instance (method & args))
```

or

```clj
(. class-or-instance method & args)
```

Both syntaxes are equivalent.

`class-or-instance` can be any arbitrary S-expression, e.g.

```clj
(.getName (let [x 1]
            (println "X is" x)
            (Thread/currentThread)))
;; -> "nREPL-session-175c7c67-d4f3-4e01-a751-ca496085f3ab"
```

Cloverage doesn't attempt to instrument the `class-or-instance` part of the `.` interop forms. Normally the `(let ...)` form above gets instrumented like:

```clj
(rw/macroexpand-all
 (inst/instrument-form
  #'inst/nop
  nil
  '(let [x 1]
     (println "X is" x)
     (Thread/currentThread))))
;; ->
(do (let* [x (do 1)]
      (do ((do println)
           (do "X is")
           (do x)))
      (do (. Thread currentThread))))
```

however, when used inside of a Java interop form, it doesn't get instrumented:

```clj
(rw/macroexpand-all
 (inst/instrument-form
  #'inst/nop
  nil
  '(.getName (let [x 1]
               (println "X is" x)
               (Thread/currentThread)))))
;; ->
(do (. (let* [x 1]
         (println "X is" x)
         (. Thread currentThread))
       getName))
```

The fix for this is actually trivial and I already have it ready, I can add it to my mega PR #304 or open a new PR once that one's merged.

**Comment by lvh** [comment]
I'll leave that up to you but I'd prefer separate if it's all the same to you. Happy to merge in any order that's convenient.

**Comment by camsaul** [comment]
Fixed in #307
